src/stories/Inputs/Upload/Upload.tsx

```diff
@@ -2,15 +2,112 @@
 import * as React from 'react';
 
 // Local Imports
-import { UploadWrapper } from 'stories/Inputs/Upload/Upload.style';
+import { UploadElement } from 'stories/Inputs/Input.style';
+import {
+  UploadWrapper,
+  UploadText,
+  UploadIconWrapper,
+  UploadElementWrapper,
+} from 'stories/Inputs/Upload/Upload.style';
+import { Icon } from 'stories/atoms/Icon/Icon';
+import { ReactComponent as UploadIcon } from 'stories/assets/upload_icon.svg';
+import { ReactComponent as CheckIcon } from 'stories/assets/check_icon.svg';
+import { ReactComponent as XIcon } from 'stories/assets/ex_icon.svg';
+import { defaultTheme } from 'themes';
+import { Loading } from 'stories/molecules/Loading/Loading';
 
-interface Props {}
+interface Props {
+  /** How should we tell the user to select a file?  */
+  selectFileText?: string;
+  /** Has the user selected at least one file? */
+  hasFile?: boolean;
+  /** Is some uploading event taking place? */
+  isUploading?: boolean;
+  /** Was the upload event successful? */
+  isUploadSuccessful?: boolean;
+  /** Did the upload event fail? */
+  hasFailed?: boolean;
+  /** What kind of files can be uploaded? */
+  accept?: string;
+  /** Can the user upload more than one file? */
+  isMultiple?: boolean;
+  /** onChange callback - will always return a FileList */
+  onChange: (fileList: FileList) => void;
```

Comment:
Like all our other `inputs`, the `onChange` handler should just pass the value back up to the calling component, which will almost always be an instance of React Hook Form's `Controller` element.

This input, instead of returning a `string`, `number`, or `boolean` value like most other inputs, returns a `fileList` object. For more on that see the JS `[File`](https://developer.mozilla.org/en-US/docs/Web/API/File) API.